code/game/objects/machinery/autolathe.dm

```diff
@@ -107,6 +107,8 @@
 	popup.set_content(dat)
 	popup.open()
 
+/obj/machinery/autolathe/update_icon()
+	icon_state = (CHECK_BITFIELD(machine_stat, PANEL_OPEN) ? "autolathe_t": "autolathe")
```

Comment:
Why is the autolathe being changed in your Warrior PR?

Reply:
atomize

Reply:
because the autolathe state change won't be reflected on punch otherwise